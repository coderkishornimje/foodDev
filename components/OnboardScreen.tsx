import {
  Text,
  StyleSheet,
  View,
  ImageBackground,
  Image,
  TouchableOpacity,
} from 'react-native';
import React, {Component} from 'react';

interface IState {
  interval: number;
  images: any[];
  texts: any[];
  childimage: any[];
  currentIndex: number;
}
interface IProps {
  navigation?: any;
}
export default class OnboardScreen extends Component<IProps, IState> {
  interval: number;

  constructor(props: IProps) {
    super(props);
    this.state = {
      images: [
        require('../images/chef1.png'),
        require('../images/pizzaboy.png'),
        require('../images/foodimg1.png'),
        require('../images/foodimg2.png'),
      ],
      texts: [
        'We have Quality Chef',
        'Swift Delivery',
        'Choose your Tasty Food',
        '10% Discount On first order',
      ],
      childimage: [require('../images/burger.png')],
      currentIndex: 0,
      
    };
  }
  componentDidMount() {
    this.interval = setInterval(() => {
      const {currentIndex, images}: any = this.state;
      const nextIndex = (currentIndex + 1) % images.length;
      this.setState({currentIndex: nextIndex});
    }, 2000);
  }
  componentWillUnmount() {
    clearInterval(this.interval);
  }
  render() {
    const {images, texts, currentIndex, childimage}: any = this.state;
    return (
      <View style={styles.container}>
        <View style={styles.child1}>
          <ImageBackground
            source={require('../images/bgColor.png')}
            resizeMode="stretch"
            style={styles.imagebg}>
            <View style={{height: '100%', width: '100%'}}>
              <Image
                style={styles.chefimg}
                //source={require('../images/chef1.png')}
                resizeMode="stretch"
                source={images[currentIndex]}
              />
            </View>
            {currentIndex == 0 ? (
              <Image
                style={styles.burgerimg}
                //source={require('../images/burger.png')}
                source={childimage[currentIndex]}
              />
            ) : null}
          </ImageBackground>
          <TouchableOpacity testID='skip-btn'
            style={styles.skipbutton}
            onPress={() => this.props.navigation.navigate('AllowLocation')}>
            <Image
              style={{height: 20, width: 20, alignSelf: 'center'}}
              source={require('../images/skip.png')}
            />
            <Text style={styles.skip}>Skip</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.child2}>
          <View style={styles.circle1}>
            <Image
              style={{height: 20, width: 20, alignSelf: 'center', top: 30}}
              source={require('../images/Vector.png')}
            />
          </View>
          <View style={styles.childpart}>
            <Text
              style={{
                fontSize: 25,
                fontWeight: 'bold',
                color: '#000000',
                alignSelf: 'center',
              }}>
              {texts[currentIndex]}
            </Text>
            <Text
              style={{fontSize: 15, fontWeight: 'bold', alignSelf: 'center'}}>
              It is a long established fact that
            </Text>
            <Text
              style={{fontSize: 15, fontWeight: 'bold', alignSelf: 'center'}}>
              A reader will be distracted
            </Text>
          </View>
          <View style={styles.pagePoint}>
            <View
              style={currentIndex == 0 ? styles.point0 : styles.point1}></View>
            <View
              style={currentIndex == 1 ? styles.point0 : styles.point1}></View>
            <View
              style={currentIndex == 2 ? styles.point0 : styles.point1}></View>
            <View
              style={currentIndex == 3 ? styles.point0 : styles.point1}></View>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: '100%',
    width: '100%',
    backgroundColor: '#fff',
  },
  child1: {
    //position: 'absolute',
    height: 600,
    width: '100%',
    backgroundColor: '#d9eda8',
  },
  child2: {
    height: 526,
    width: 550,
    backgroundColor: '#fff',
    position: 'absolute',
    borderRadius: 526 / 2,
    top: 500,
    left: -70,
    alignSelf: 'center',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  imagebg: {
    height: '100%',
    width: '100%',
    flexDirection: 'row',
  },
  chefimg: {
    height: '100%',
    width: '100%',
    //marginTop: 30,
  },
  burgerimg: {
    height: 120,
    width: 130,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    right: 50,
    bottom: 150,
  },
  mainCircle: {
    // height:4200,
    // borderRadius:75,
    // backgroundColor:'red',
  },
  circle1: {
    position: 'absolute',
    backgroundColor: 'red',
    height: 80,
    width: 80,
    borderRadius: 50,
    top: -32,
  },
  childpart: {
    bottom: 100,
  },
  pagePoint: {
    flexDirection: 'row',
    bottom: 40,
  },
  point0: {
    borderWidth: 2,
    borderColor: '#94CD00',
    backgroundColor: '#DF201F',
    height: 15,
    width: 15,
    borderRadius: 100,
    marginRight: 10,
  },
  point1: {
    borderWidth: 2,
    borderColor: '#94CD00',
    // backgroundColor:'#DF201F',
    height: 15,
    width: 15,
    borderRadius: 100,
    marginRight: 10,
  },
  skip: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#000000',
    marginLeft: 10,
  },
  skipbutton: {
    backgroundColor: 'white',
    right: 0,
    left: 310,
    //bottom: 450,
    top:-550,
    flexDirection: 'row',
    padding: 5,
    borderRadius: 10,
  },
});
