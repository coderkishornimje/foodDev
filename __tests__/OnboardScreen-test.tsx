import {
  fireEvent,
    render,

  } from '@testing-library/react-native';
  import renderer from 'react-test-renderer';
import OnboardScreen from '../components/OnboardScreen';

describe('home component', () => {
    it('renders correctly', () => {
      renderer.create(<OnboardScreen />);
    });
    it('to check app render correctly', () => {
      const appRender = render(<OnboardScreen />);
      expect(appRender).toMatchSnapshot();
      expect(appRender).toBeDefined();
    });
    it('check skip button',()=>{
      const navigation={navigate:jest.fn()}
      const {getByTestId}=render(<OnboardScreen navigation={navigation}/>);
      const skipoption=getByTestId('skip-btn');
      fireEvent.press(skipoption);
      expect(navigation.navigate).toHaveBeenCalledTimes(1);
    })
    it('check settimeout home page',()=>{
        jest.useFakeTimers();
        const appRender = render(<OnboardScreen />);
         jest.advanceTimersByTime(2000);
         jest.advanceTimersByTime(2000);
         jest.advanceTimersByTime(2000);
        expect(appRender).toBeDefined();
    });

})  