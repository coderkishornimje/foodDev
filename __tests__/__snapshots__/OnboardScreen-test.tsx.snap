// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component to check app render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#d9eda8",
        "height": "70%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "flexDirection": "row",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="stretch"
        source={
          {
            "testUri": "../../../images/bgColor.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={
            {
              "testUri": "../../../images/chef1.png",
            }
          }
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <Image
        source={
          {
            "testUri": "../../../images/burger.png",
          }
        }
        style={
          {
            "alignItems": "center",
            "bottom": 150,
            "height": 120,
            "justifyContent": "center",
            "position": "absolute",
            "right": 50,
            "width": 130,
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "bottom": 430,
          "flexDirection": "row",
          "left": 300,
          "opacity": 1,
          "padding": 5,
          "right": 0,
        }
      }
      testID="skip-btn"
    >
      <Image
        source={
          {
            "testUri": "../../../images/skip.png",
          }
        }
        style={
          {
            "alignSelf": "center",
            "height": 20,
            "width": 20,
          }
        }
      />
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginLeft": 10,
          }
        }
      >
        Skip
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 263,
        "flexDirection": "column",
        "height": 526,
        "justifyContent": "center",
        "left": -60,
        "position": "absolute",
        "top": 370,
        "width": 526,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "red",
          "borderRadius": 50,
          "height": 80,
          "position": "absolute",
          "top": -32,
          "width": 80,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../images/Vector.png",
          }
        }
        style={
          {
            "alignSelf": "center",
            "height": 20,
            "top": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      style={
        {
          "bottom": 100,
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#000000",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        We have Quality Chef
      </Text>
      <Text
        style={
          {
            "alignSelf": "center",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        It is a long established fact that
      </Text>
      <Text
        style={
          {
            "alignSelf": "center",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        A reader will be distracted
      </Text>
    </View>
    <View
      style={
        {
          "bottom": 40,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#DF201F",
            "borderColor": "#94CD00",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": 15,
            "marginRight": 10,
            "width": 15,
          }
        }
      />
      <View
        style={
          {
            "borderColor": "#94CD00",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": 15,
            "marginRight": 10,
            "width": 15,
          }
        }
      />
      <View
        style={
          {
            "borderColor": "#94CD00",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": 15,
            "marginRight": 10,
            "width": 15,
          }
        }
      />
      <View
        style={
          {
            "borderColor": "#94CD00",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": 15,
            "marginRight": 10,
            "width": 15,
          }
        }
      />
    </View>
  </View>
</View>
`;
